hive-metastore/src/test/java/org/apache/iceberg/hive/HiveMetastoreExtension.java

```diff
@@ -40,16 +40,10 @@ private HiveMetastoreExtension(String databaseName, Map<String, String> hiveConf
 
   @Override
   public void beforeAll(ExtensionContext extensionContext) throws Exception {
-    metastore = new TestHiveMetastore();
-    HiveConf hiveConfWithOverrides = new HiveConf(TestHiveMetastore.class);
-    if (hiveConfOverride != null) {
-      for (Map.Entry<String, String> kv : hiveConfOverride.entrySet()) {
-        hiveConfWithOverrides.set(kv.getKey(), kv.getValue());
-      }
-    }
+    metastore = new TestHiveMetastore(hiveConfOverride);
 
-    metastore.start(hiveConfWithOverrides);
-    metastoreClient = new HiveMetaStoreClient(hiveConfWithOverrides);
+    metastore.start(new HiveConf(TestHiveMetastore.class));
+    metastoreClient = new HiveMetaStoreClient(metastore.hiveConf());
```

Comment:
Why is this change?

Reply:
`METASTORE_TRY_DIRECT_SQL` is hard coded to false in `TestHiveMetastore::initConf`. This change is to make it possible to override the config in our test.

Reply:
Could you please try to set this globally, and see if there is an issue with it? This is just test code, and I would like to try to keep it as simple as possible.

Reply:
OK I'll try setting the default value to true, if everything is fine, I'll revert this change

Reply:
Enabling direct SQL globally seems to fix the [weird test case](https://github.com/apache/iceberg/blob/af99da6c2daa8e53876638c839b0f729f2f3a86e/spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAddFilesProcedure.java#L741). I did some debug and believe it's because direct SQL gets the partition filter pushdown correct and retrieves the corresponding partitions. Should I update the test cases here, or leave it to another PR?